sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -276,10 +293,55 @@ public String toString(Charset charSet) {
      * @param clazz representing the type of the Object.
      * @param serializer to use deserialize data into type.
      * @param <T> Generic type that the data is deserialized into.
-     * @throws NullPointerException if {@code clazz} or {@code serializer} is null.
+     * @throws NullPointerException If {@code clazz} or {@code serializer} is null.
      * @return The {@link Object} of given type after deserializing the bytes.
      */
     public  <T> Mono<T> toObjectAsync(Class<T> clazz, ObjectSerializer serializer) {
         return Mono.fromCallable(() -> toObject(clazz, serializer));
     }
+
+    /**
+     * Deserialize the bytes into the {@link Object} of given type by using json serializer which is available in
+     * classpath. The serializer must implement {@link JsonSerializer} interface. A singleton instance of
+     * {@link JsonSerializer} is kept for this class to use.
+     *
+     * @param clazz representing the type of the Object.
+     * @param <T> Generic type that the data is deserialized into.
+     * @return The {@link Object} of given type after deserializing the bytes.
+     */
+    public <T> T toObject(Class<T> clazz) {
```

Comment:
Should we rename this API to something JSON specific? `toJsonObject`?

@JonathanGiles @srnagar

Reply:
This might force us to add other API for other formats.

Reply:
JsonObject has a different meaning. So, I think `toObject()` is fine as we will return a strongly-typed user-defined object here.